js/CobrowseIO.js

```diff
@@ -4,6 +4,84 @@ const NativeEventEmitter = require('react-native').NativeEventEmitter
 
 const emitter = new NativeEventEmitter(CobrowseIONative)
 
+class Session {
+  constructor (session) {
+    this._session = session || {}
+  }
+
+  get id () {
+    return this._session.id
+  }
+
+  get code () {
+    return this._session.code
+  }
+
+  get state () {
+    return this._session.state
+  }
+
+  get full_device () {
+    return this._session.full_device
+  }
+
+  get remote_control () {
+    return this._session.remote_control
+  }
+
+  get agent () {
+    return this._session.agent
+  }
+
+  activate () {
+    return CobrowseIONative.activateSession().then((session) => {
+      this._session = session
+    })
+  }
+
+  end () {
+    return CobrowseIONative.endSession()
+  }
+
+  hasAgent () {
+    return CobrowseIONative.sessionHasAgent()
+  }
+
+  isActive () {
+    return CobrowseIONative.sessionIsActive()
+  }
+
+  isAuthorizing () {
+    return CobrowseIONative.sessionIsAuthorizing()
+  }
+
+  isPending () {
+    return CobrowseIONative.sessionIsPending()
+  }
+
+  isEnded () {
+    return CobrowseIONative.sessionIsEnded()
+  }
+
+  setFullDevice (state) {
+    return CobrowseIONative.sessionSetFullDevice(state)
```

Comment:
When we spoke on Thursday I suggested you could implement these using a generic update(...) method tat takes an object, e.g. setFullDevice would be:

```
setFullDevice (state) {
     return CobrowseIONative.updateSession({ full_device: true })
}
```

This will save a lot of repeated code for all the setters that would otherwise be needed on iOS and Android.

Reply:
I was going to call the generic method in the Java code, but this makes better sense since the JS method is already wrapping a Java method.